other/scoring_algorithm.py

```diff
@@ -21,39 +22,49 @@
 Thus the weights for each column are as follows:
 [0, 0, 1]
 
->>> procentual_proximity([[20, 60, 2012],[23, 90, 2015],[22, 50, 2011]], [0, 0, 1])
+>>> score([[20, 60, 2012],[23, 90, 2015],[22, 50, 2011]], [0, 0, 1])
 [[20, 60, 2012, 2.0], [23, 90, 2015, 1.0], [22, 50, 2011, 1.3333333333333335]]
+>>> score([[20, 60, 2012],[23, 90, 2015],[22, 50, 2011]], [0, 0, 1], 'scores')
+[2.0, 1.0, 1.3333333333333335]
+>>> score_columns([[20, 60, 2012],[23, 90, 2015],[22, 50, 2011]], [0, 2], [0, 0, 1])
+[[20, 2012, 1.25], [23, 2015, 1.0], [22, 2011, 0.33333333333333337]]
 """
 
 
-def procentual_proximity(source_data: list, weights: list) -> list:
-
-    """
-    weights - int list
-    possible values - 0 / 1
-    0 if lower values have higher weight in the data set
-    1 if higher values have higher weight in the data set
+def score(source_data: list, weights: list, *args) -> list:
```

Comment:
I liked the old function name better because it documented the analysis that was being run.  It was specific whereas there are a million and one kinds of scoring that can be done on a dataset and the new function name forces me to read the docs or the implementation to know.